Tests broken with google.golang.org/protobuf v1.33.0

In [/desc/builder/builder_test.go](https://github.com/jhump/protoreflect/blob/7195054027de0de09df8b00cfa8ecc5775201c36/desc/builder/builder_test.go)
TestInvalid/group_in_proto3
TestInvalid/default_value_in_proto3

```
$ go test ./desc/builder/
--- FAIL: TestInvalid (0.00s)
    --- FAIL: TestInvalid/group_in_proto3 (0.00s)
        builder_test.go:1733: unexpected error: want "invalid group: invalid under proto2 semantics", got "proto: message field \"Foo.bar\" is an invalid group: invalid under proto3 semantics"
    --- FAIL: TestInvalid/default_value_in_proto3 (0.00s)
        builder_test.go:1733: unexpected error: want "invalid default: cannot be specified under proto3 semantics", got "proto: message field \"Foo.foo\" has invalid default: cannot be specified with implicit field presence"
FAIL
FAIL	github.com/jhump/protoreflect/desc/builder	0.074s
FAIL
```

NB: google.golang.org/protobuf@v1.33.0 resolve CVE-2024-24786
https://groups.google.com/g/golang-announce/c/ArQ6CDgtEjY
https://github.com/protocolbuffers/protobuf-go/releases/tag/v1.33.0

--- a/desc/builder/builder_test.go
+++ b/desc/builder/builder_test.go
@@ -1661,9 +1661,7 @@ func TestInvalid(t *testing.T) {
 						NewMessage("Foo").AddField(NewGroupField(NewMessage("Bar"))),
 					)
 			},
-			// NB: This is the actual error message returned by the protobuf runtime. It is
-			//     misleading since it says proto2 instead of proto3.
-			expectedError: "invalid group: invalid under proto2 semantics",
+			expectedError: "invalid group: invalid under proto3 semantics",
 		},
 		{
 			name: "default value in proto3",
@@ -1673,7 +1671,7 @@ func TestInvalid(t *testing.T) {
 						NewMessage("Foo").AddField(NewField("foo", FieldTypeString()).SetDefaultValue("abc")),
 					)
 			},
-			expectedError: "invalid default: cannot be specified under proto3 semantics",
+			expectedError: "invalid default: cannot be specified with implicit field presence",
 		},
 		{
 			name: "extension tag outside range",

--- a/go.mod
+++ b/go.mod
@@ -4,11 +4,11 @@ go 1.18
 
 require (
 	github.com/bufbuild/protocompile v0.8.0
-	github.com/golang/protobuf v1.5.3
+	github.com/golang/protobuf v1.5.4
 	github.com/jhump/gopoet v0.1.0
 	github.com/jhump/goprotoc v0.5.0
 	google.golang.org/grpc v1.57.1
-	google.golang.org/protobuf v1.31.1-0.20231027082548-f4a6c1f6e5c1
+	google.golang.org/protobuf v1.33.0
 )
 
 require (

